pandas/tseries/resample.py

```diff
@@ -14,7 +21,560 @@
+    __inv__ = _make_deprecated_unary(lambda x: -x)
+
+    def __getattr__(self, attr):
+        if attr in self._internal_names_set:
+            return object.__getattribute__(self, attr)
+        if attr in self._attributes:
+            return getattr(self.groupby, attr)
+        if attr in self.obj:
+            return self[attr]
+
+        if attr in self._deprecated_invalids:
+            raise ValueError(".resample() is now a deferred operation\n"
+                             "\tuse .resample(...).mean() instead of "
+                             ".resample(...)\n"
+                             "\tassignment will have no effect as you "
+                             "are working on a copy")
+        if attr not in self._deprecated_valids:
+            self = self._deprecated()
+        return object.__getattribute__(self, attr)
+
+    def __setattr__(self, attr, value):
+        if attr not in self._deprecated_valids:
+            raise ValueError("cannot set values on {0}".format(
+                self.__class__.__name__))
+        object.__setattr__(self, attr, value)
+
+    def __setitem__(self, attr, value):
+        raise ValueError("cannot set items on {0}".format(
+            self.__class__.__name__))
+
+    def _convert_obj(self, obj):
+        """
+        provide any conversions for the object in order to correctly handle
+
+        Parameters
+        ----------
+        obj : the object to be resampled
+        """
+        obj = obj.consolidate()
+        return obj
+
+    def _get_binner_for_time(self):
+        raise AbstractMethodError(self)
+
+    def _set_binner(self):
+        """ setup our binners """
+        self.binner, self.grouper = self._get_binner()
+
+    def _get_binner(self):
+        """
+        create the BinGrouper, assume that self.set_grouper(obj)
+        has already been called
+        """
+
+        binner, bins, binlabels = self._get_binner_for_time()
+        bin_grouper = BinGrouper(bins, binlabels)
+        return binner, bin_grouper
+
+    def aggregate(self, arg, *args, **kwargs):
+
+        self._set_binner()
+        result, how = self._aggregate(arg, *args, **kwargs)
+        if result is None:
+            return self._groupby_and_aggregate(self.grouper,
+                                               arg,
+                                               *args,
+                                               **kwargs)
+
+        return result
+
+    agg = aggregate
+    apply = aggregate
+
+    def transform(self, arg, *args, **kwargs):
+        """
+        Provide a transformation on the Resampler
+        Return the same size as input
+        """
+        return self._selected_obj.groupby(self.groupby).transform(
+            arg, *args, **kwargs)
+
+    def _downsample(self, f):
+        raise AbstractMethodError(self)
+
+    def _upsample(self, f, limit=None):
+        raise AbstractMethodError(self)
+
+    def _gotitem(self, key, ndim, subset=None):
```

Comment:
out of curiosity, is there a reason this is called `gotitem` (and not `getitem`)?